pkg/neotest/basic.go

```diff
@@ -114,13 +114,24 @@ func (e *Executor) NewAccount(t *testing.T) Signer {
 	return NewSingleSigner(acc)
 }
 
-// DeployContract compiles and deploys contract to bc.
+// DeployContract compiles and deploys contract to bc. It also checks that
+// precalculated contract hash matches the actual one.
 // data is an optional argument to `_deploy`.
 // Returns hash of the deploy transaction.
 func (e *Executor) DeployContract(t *testing.T, c *Contract, data interface{}) util.Uint256 {
 	tx := e.NewDeployTx(t, e.Chain, c, data)
 	e.AddNewBlock(t, tx)
 	e.CheckHalt(t, tx.Hash())
+
+	// Check that precalculated hash matches the real one.
```

Comment:
Can it not be the case?

Reply:
Hash is calculated by `state.CreateContractHash(sender util.Uint160, checksum uint32, name string)` for both pre-calculated and real hashes. The contract is deployed from the committee address each time, so the `sender` is the same for both cases. Also, NEX and manifest are the same too, so `checksum` and `name` should match.

So I think it can't be the case. But at the same time, some tests have this check, and it costs nothing to preserve it.

So should we remove it?

Reply:
We're testing here and we had similar checks already, it doesn't cost a lot.